### DISCUSSION TITLE
Passing Cache-Control headers forward

### DISCUSSION THREAD

**Discussion by akshatmittal** [vercel/vercel#4144] [General] [answered]
At the moment, I add specific `Cache-Control` headers to my Serverless endpoints as well as page endpoints with `getServerSideProps`. This is working well and good with ~Zeit CDN~ Vercel Edge Network caching the responses for the said amount of time along with `stale-while-revalidate`, but I want to pass these headers forward since I use CloudFlare and would like to cache some of that data at the CloudFlare Edge to free up Vercel bandwidth and speed up page load times.

Is there any way to do that?

**Comment by coetry** [comment, marked as answer]
Hi @akshatmittal layering another proxy in front of ours actually will not speed up your pages, but will create more complexity and introduce caching errors that will be out of our control. If something is cached on Cloudflare, we will not able able to purge that cache on a new deploy. Furthermore, you'll be introducing more networking hops that your page needs to route through.

This is not a setup that we support and we can assure you that depending on our Edge alone will be more than sufficient :smile:

**Reply by akshatmittal** [reply]
I actually need to use CloudFlare because I'm using some other features such as Firewall and WAF.

I only intend to cache the dynamically created pages on the CloudFlare Edge and use Firewall on the API. It's doing well and good right now too, but some control via the cache headers would be great.

I understand it's not a supported use-case, thanks for the help!

**Reply by arguiot** [reply]
@coetry I have the exact same issue here. CloudFlare has a lot of great options like Firewall, email protection, auto-minify, etc. that aren't there on Vercel Edge Network, so is there a way to bypass Vercel Edge Network to directly point the domain towards Vercel servers?

```
Before:

User -> Vercel Edge -> Deployment

What I want:

User -> Cloudflare -> Deployment

If not possible:

User -> Cloudflare -> Vercel Edge -> Deployment
```
